server/src/test/java/org/elasticsearch/cluster/routing/OperationRoutingTests.java

```diff
@@ -523,13 +523,11 @@ public void testAdaptiveReplicaSelection() throws Exception {
                 new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS));
         opRouting.setUseAdaptiveReplicaSelection(true);
         List<ShardRouting> searchedShards = new ArrayList<>(numShards);
-        Set<String> selectedNodes = new HashSet<>(numShards);
         TestThreadPool threadPool = new TestThreadPool("testThatOnlyNodesSupportNodeIds");
         ClusterService clusterService = ClusterServiceUtils.createClusterService(threadPool);
         ResponseCollectorService collector = new ResponseCollectorService(clusterService);
-        Map<String, Long> outstandingRequests = new HashMap<>();
```

Comment:
Some notes:
* I had to make loosen the checks before the test would pass. I tracked down the surprising behavior to `IndexShardRoutingTable#adjustStats`, which can give a big boost to slower nodes in the ranking. I plan to open a follow-up issue proposing more tests and a potential fix.
* When a 'winner' node is selected, we update the local 'outstanding requests' map to increment its value by 1. This only affects the local copy, it doesn't update global statistics. While this has a purpose in the ARS logic, it doesn't make sense in the context of these tests. So I removed the shared `outstandingRequests` map.